### DISCUSSION TITLE
DevTools - export props interfaces

### DISCUSSION THREAD

**Discussion by UgRoss** [TanStack/query#2658] [Ideas] [answered]
Hello everyone! 👋

Would it be possible to export props interfaces from `react-query/devtools` (`interface DevtoolsOptions` / `interface DevtoolsPanelOptions`)? 
We are trying to create a reusable `QueryClientProvider` wrapper that also includes `ReactQueryDevtools` but it's not possible to import props interface from `react-query/devtools`.

Example:
```typescript
import { QueryClient, QueryClientProvider } from 'react-query';
import { ReactQueryDevtools, DevtoolsOptions } from 'react-query/devtools'; 
// ↑ ❌ Module '"react-query/devtools"' has no exported member 'DevtoolsOptions'.

export interface ReactQueryClientProviderProps {
  client: QueryClient,
  children: React.ReactChild,
  devToolsOptions?: DevtoolsOptions,
}

export const ReactQueryClientProvider: React.FC<ReactQueryClientProviderProps> = () => {
  // ...
}
```

P.S. In case it's possible to do I can open a PR for this.
Thanks!

**Comment by TkDodo** [comment, marked as answer]
the best way is to extract the props:

```
type DevtoolsOptions = React.ComponentProps<typeof ReactQueryDevtools>
```

**Reply by UgRoss** [reply]
oh, got it 👍
Thank you for the quick response!
